src/utils/decorators/baseDecoratorExtractor.ts

```diff
@@ -58,22 +72,25 @@ export abstract class DecoratorExtractor {
         return methodInfo;
     }
 
-    createGenezioClassInfo(className: string, file: string, commentText: string): ClassInfo {
+    static createGenezioClassInfo(className: string, file: string, commentText: string): ClassInfo {
         const genezioDeployArguments = commentText
             .split("genezio: deploy")[1]
             .split(" ")
             .filter((arg) => arg !== "");
         let classType: string = "jsonrpc";
-        switch (genezioDeployArguments[0]) {
-            case "http": {
-                classType = "http";
-                break;
-            }
-            case "cron": {
-                classType = "cron";
-                break;
+        if (genezioDeployArguments.length > 0) {
```

Comment:
this is also not needed, if the array is empty, the element at index 0 would be undefined, meaning it would not satisfy any cases of the switch statement

Reply:
the problem is the genezio deploy arguments with this commentText would be undefined and it would throw and error arguments[0]